src/artm/core/processor.cc

```diff
@@ -184,7 +185,8 @@ static void RegularizeAndNormalizeTheta(int inner_iter, const Batch& batch, cons
       continue;
     }
 
-    regularizer->RegularizeTheta(batch, model_config, inner_iter, tau, theta);
+    regularizer->RegularizeTheta(batch, model_config, inner_iter, tau, theta,
+                                 merger.GetLatestTopicModel(model_config.name()));
```

Comment:
The code in processor should use merger.GetLatestTopicModel() function only ONCE for each (batch, model) pair, because the entire batch must be handled with a single Phi matrix.

Calling this method again might result in a different topic model used by regularizers comparing to the other code in the processor. This feels really bad (think of adding/removing tokens and topics... crazy things might happen to regularizer in this case).

I think for now you should revert this change since it conflicts with my re-work in Processor.cc. Lets re-introduce it after my pull request is in the master.